control/ctrlplot.py

```diff
@@ -0,0 +1,74 @@
+# ctrlplot.py - utility functions for plotting
+# Richard M. Murray, 14 Jun 2024
+#
+# Collection of functions that are used by various plotting functions.
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+from . import config
+
+__all__ = ['suptitle']
+
+
+def suptitle(
+        title, fig=None, frame='axes', **kwargs):
+    """Add a centered title to a figure.
+
+    This is a wrapper for the matplotlib `suptitle` function, but by
+    setting ``frame`` to 'axes' (default) then the title is centered on the
+    midpoint of the axes in the figure, rather than the center of the
+    figure.  This usually looks better (particularly with multi-panel
+    plots), though it takes longer to render.
+
+    Parameters
+    ----------
+    title : str
+        Title text.
+    fig : Figure, optional
+        Matplotlib figure.  Defaults to current figure.
+    frame : str, optional
+        Coordinate frame to use for centering: 'axes' (default) or 'figure'.
+    **kwargs : :func:`matplotlib.pyplot.suptitle` keywords, optional
+        Additional keywords (passed to matplotlib).
+
+    """
+    rcParams = config._get_param('freqplot', 'rcParams', kwargs, pop=True)
+
+    if fig is None:
+        fig = plt.gcf()
+
+    if frame == 'figure':
+        with plt.rc_context(rcParams):
+            fig.suptitle(title, **kwargs)
+
+    elif frame == 'axes':
+        # TODO: move common plotting params to 'ctrlplot'
```

Comment:
This same TODO comment is already freqplot.py, and it is not clear what the comment means here (in ctrlplot). Should the TODO comment be deleted here in favor of the one in freqplot.py ?

Reply:
Reading it again, I think this TODO comment is to be able to
```py
rcParams = config._get_param('ctrlplot', 'rcParams', rcParams)
```
If so, then I understand the motivation, and OK to keep both TODO comments.

Reply:
I've got another PR coming that will address both TODOs.  Doing things in stages so that everything is in (slightly) smaller chunks.

Reply:
Awesome. I am almost done reviewing this one.